engine-api/src/server.rs

```diff
@@ -159,6 +168,27 @@ async fn update_module_cfg(
     Ok(())
 }
 
+async fn event_monitor_handler(
+    State(ctx): State<EngineAPIContext>,
+    ws: WebSocketUpgrade,
+) -> Response {
+    let mut bus_receiver = ctx.bus.get_receiver();
+
+    // This closure reads events from the bus receiver and sends them into the socket
+    let handle_socket = |mut socket: WebSocket| async move {
+        while let Ok(event) = bus_receiver.recv().await {
+            if let Ok(json) = serde_json::to_string(&*event) {
```

Comment:
This error is ignored, which could lead to infinite loop. You could argue `serde_json::to_string` will never fail, but in that case an `expect` panic would be more appropriate.
I'd log an error and return.